### PR TITLE
fetch api doc

### PR DESCRIPTION
**Pull request by dmzmk**

### CODE REVIEW

**Comment by rdowinton** [comment]
Looks good, but should it not fall under the 'Working with HTTPS' section? The fetch API is the main focus of that section, even though there is very little info on how to use it.

**Comment by dmzmk** [comment]
True, probably I should merge this sections.

**Comment by qrilka** [comment]
@rdowinton I don't think that we should merge those sections. HTTPS is the main motivation for the fetch API but you could use it not only for HTTPS

File: crawlera.rst [hunk 1]
```diff
@@ -49,6 +50,31 @@ The caveat of Crawlera’s HTTP proxy interface is that it doesn’t support the
 Note that it’s not recommended to use the *fetch API* with the Crawlera middleware. Another option is the :ref:`x-crawlera-use-https` header.
 
 
+Fetch API
+=========
```
Comment (pablohoffman) [inline crawlera.rst]: i'd put a space here for legibility and consistency

File: crawlera.rst [hunk 2]
```diff
@@ -49,6 +50,31 @@ The caveat of Crawlera’s HTTP proxy interface is that it doesn’t support the
...
+| Crawlera provides HTTP API interface for URL fetching.
```
Comment (pablohoffman) [inline crawlera.rst]: We need to write this better. @rdowinton can you take a pass?

It reads like a bullet point but it should be a paragraph describing what the fetch API is, to be consistent with the rest of the document.

File: crawlera.rst [hunk 3]
```diff
@@ -49,6 +50,31 @@ The caveat of Crawlera’s HTTP proxy interface is that it doesn’t support the
...
+============== ========== ========================================================== ========================================================
+Field          Mandatory  Description                                                Example
+============== ========== ========================================================== ========================================================
+url            yes        URL to fetch                                               `http://crawlera.com`
+headers        no         additional headers which will be used in outgoing request  `Header1:Value1;Header2:Value2`
+============== ========== ========================================================== ========================================================
+
+.. warning::
+
+    It's always a good idea to encode field values, otherwise fetched result can be incorrect.
```
Comment (pablohoffman) [inline crawlera.rst]: It shouldn't be just a good idea, it should be a requirement.

**Comment by pablohoffman** [comment]
agree with @qrilka, it warrants a section of it self (maybe "HTTP API"  instead of "Fetch API"?) and it can be pointed from the HTTPS question.

**Comment by rdowinton** [comment]
@pablohoffman @qrilka sounds good, as you suggested Pablo we should keep the HTTPS section but shorten it and point to the fetch API part. I'll update the wording to make it more legible.

**Comment by rdowinton** [comment]
I've updated the PR. Here's a summary of the changes I've made:

* Added introductory paragraph to fetch API section.
* Reworded some of the existing text.
* Changed the encoding message from a warning to a note. While it's important, I think warnings should be reserved for anything unexpected and in this case most people will expect to have to encode the values.
* Moved fetch API example from 'Working with HTTPS' section to 'Fetch API' section.

Please let me know your thoughts and I'll close if there aren't any changes required.

**Pull request merged**